src/main/java/net/neoforged/neoforge/transfer/transaction/TransactionManagerImpl.java

```diff
@@ -0,0 +1,227 @@
+package net.neoforged.neoforge.transfer.transaction;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+
+@ApiStatus.Internal
+public class TransactionManagerImpl {
+    public static final ThreadLocal<TransactionManagerImpl> MANAGERS = ThreadLocal.withInitial(TransactionManagerImpl::new);
+
+    private final Thread thread = Thread.currentThread();
+    private final ArrayList<TransactionImpl> stack = new ArrayList<>();
+    private final ArrayList<Transaction.OuterCloseCallback> outerCloseCallbacks = new ArrayList<>();
```

Comment:
Is there a reason these two need to be explicitly `ArrayList` instead of say just `List`?